packages/router-plugin/src/core/code-splitter/compilers.ts

```diff
@@ -142,7 +142,7 @@ export function compileCodeSplitReferenceRoute(opts: ParseAstOptions) {
 
                           programPath.pushContainer('body', [
                             template.statement(
-                              `function DummyComponent() { return null }`,
+                              `export function TSR_DummyComponent() { return null }`,
```

Comment:
This forces Fast Refresh to process the original route file even if we end up extracting the HMR passing conditions out of it.

Reply:
Marking the route as an HMR boundary without handling HMR will lead to stale updates. Instead using the patch provided by @schiller-manuel in discord is good and both mark the module as an self accepting without fast refresh code and handle the update.